### PR TITLE
ECO-323: Move code to this repo.

### PR DESCRIPTION
**Pull request by zie1ony**
### Overview
* Move ERC20 code to this repo.
* Refactor contract crate.
* Write tests using testing framework.

### Which JIRA ticket does this PR relate to?
https://casperlabs.atlassian.net/browse/ECO-323

### Complete this checklist before you submit this PR
- [x] This PR contains no more than 200 lines of code, excluding test code.
- [x] This PR meets [CasperLabs coding standards](https://casperlabs.atlassian.net/wiki/spaces/EN/pages/16842753/Coding+Standards).
- [x] If this PR adds a new feature, it includes tests related to this feature.
- [x] You assigned one person to review this PR.
- [x] Your GitHub account is linked with our [Drone CI](https://drone-auto.casperlabs.io/) system. This is necessary to run tests on this PR.
- [x] Do not forget to run `bors r+` if GitHub policy is not enforced, e.g. when merging into another feature branch. It may be omitted under some circumstances if this PR intentionally assumes that integration tests will fail but will be fixed with the future PRs.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  - "rustup target add wasm32-unknown-unknown"
+  - "make test"
+  image: "casperlabs/buildenv:latest"
+  when:
+    event:
+    - pull_request
+    status:
+    - success
+  depends_on:
+  - lint-pr
```
Comment (TomVasile) [inline .drone.yml]: these can probably run in parallel?

File: .drone.yml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  commands:
+  - "rustup target add wasm32-unknown-unknown"
+  - "make test"
+  image: "casperlabs/buildenv:latest"
+  when:
+    branch:
+    - staging
+    - trying
+  depends_on:
+  - lint-bors
```
Comment (TomVasile) [inline .drone.yml]: same here?

File: .drone.yml [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+
+volumes:
+- name: docker_sock
+  host:
+    path: "/var/run/docker.sock"
+- name: temp
+  host:
+    path: "/tmp"
```
Comment (TomVasile) [inline .drone.yml]: this isn't being used from what i can see.

**Comment by TomVasile** [comment]
we can probably simplify the `.drone.yml` as well to something like this for the steps and do away with the duplication:

```
steps:
- name: lint-pr
  commands:
  - "make lint"
  image: "casperlabs/buildenv:latest"
  when:
    event:
    - pull_request
    - push
    branch:
    - staging
    - trying
    - master
```

File: .drone.yml (hunk 3 above)
Comment (arcolife) [reply .drone.yml]: isn't this for the `image` in steps ?

File: .drone.yml (hunk 1 above)
Comment (arcolife) [reply .drone.yml]: don't think running tests is worth it if lint fails.

Comment (TomVasile) [reply .drone.yml]: fair but it will save time

File: .drone.yml (hunk 3 above)
Comment (TomVasile) [reply .drone.yml]: this is for using host volume mounts. Theres no point to mount them if they are not being used.

**Comment by TomVasile** [comment]
bors try

**Comment by TomVasile** [comment]
bors try

**Comment by TomVasile** [comment]
@arcolife I simplified the drone steps further (down to just lint and test once) and added the bors config file. This should be set.

**Pull request merged**